src/qtcore/qml/elements/QtQuick/Rectangle.js

```diff
@@ -19,6 +19,7 @@ registerQmlType({
         this.dom.style.borderRadius = newVal + "px";
     });
     this.border.colorChanged.connect(this, function(newVal) {
+        this.dom.style.borderWidth = this.border.width == 0 ? "1px" : this.border.width + "px";
         this.dom.style.borderColor = newVal;
         this.dom.style.borderStyle = this.border.width == 0 || newVal == "transparent"
```

Comment:
On this line, even when `this.dom.style.borderWidth` is set to `"1px"`, when `this.border.width == 0`, there is no border because of the `this.dom.style.borderStyle` being set to `"none"`.